cmd/github.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+)
+
+// lookupGitHubToken will try to find a GitHub access token that can then be used for various API calls but also cloning of private repositories.
+func lookupGitHubToken(cctx *cli.Context) (string, error) {
+	log.Print("Looking for a GitHub token")
+	token := cctx.String("github-token")
+	if token != "" {
+		log.Print("Using GitHub token provided via flag")
+		return token, nil
+	}
+	// First try: Check if it's in the environment. This can override everything!
+	token = os.Getenv("GITHUB_TOKEN")
+	if token != "" {
+		log.Print("Using GitHub token provided via environment variable")
+		return token, nil
+	}
+
+	// Next, check if the user has gh installed and *it* has a token set:
+	var data bytes.Buffer
+	var errData bytes.Buffer
+	ghPath, err := exec.LookPath("gh")
+	if err == nil {
+		cmd := exec.CommandContext(cctx.Context, ghPath, "auth", "token")
+		cmd.Stdout = &data
+		cmd.Stderr = &errData
+		err = cmd.Run()
+		if err != nil {
+			log.Printf("Querying gh for an access token failed: %s", errData.String())
+			return "", fmt.Errorf("lookup in gh failed: %w", err)
+		}
+		log.Print("Using GitHub token provided via gh")
+		return strings.TrimSpace(data.String()), nil
+	}
+
+	return "", nil
```

Comment:
```suggestion
	ghPath, err := exec.LookPath("gh")
	if err != nil {
		return "", nil
	}
	
	cmd := exec.CommandContext(cctx.Context, ghPath, "auth", "token")
	cmd.Stdout = &data
	cmd.Stderr = &errData
	
	if err := cmd.Run(); err != nil {
		log.Printf("Querying gh for an access token failed: %s", errData.String())
		return "", fmt.Errorf("lookup in gh failed: %w", err)
	}
	
	log.Print("Using GitHub token provided via gh")
	return strings.TrimSpace(data.String()), nil
```

I just inversed the conditions here to be more consistent with the rest of the project.

Should this condition return an error?

```
	ghPath, err := exec.LookPath("gh")
	if err != nil {
		return "", nil
	}
```

I'm fine with returning an error here saying something like, "Expected a `--github-token` flag, `GITHUB_TOKEN` environment variable, or a configured GitHub CLI (`$XDG_CONFIG_HOME/gh` or `%appdata%/gh`)"

Reply:
Personally, I think it would be nicer to have that as fall-through response: `return "", fmt.Errorf("expected ...")`